src/core/Math.re

```diff
@@ -5,9 +5,10 @@ let sqf = x => x *. x;
 // Exponents.
 let rec pow = (~base, ~exp) => {
   switch (exp) {
-  | 0 => 1
-  | 1 => base
-  | _ => base * pow(~base, ~exp=exp - 1)
+  | 0 => 1.
+  | 1 => Utils.foi(base)
+  | n when n < 0 => 1. /. Utils.foi(base) *. pow(~base, ~exp=exp + 1)
```

Comment:
Handle negative exponents!

Reply:
Noice!  But question, do you think you'll ever need fractional exponent support?  Does reason's type casting let you enforce that Math.pow only supports integer exponents?

Reply:
This more has to do with supporting operations like calculating the value of the universal gravitational constant `G`, which is 6.67 * 10^-11. AFAIK, 10^-11 is equivalent to 1/10^11, which is also equivalent to 1/10 * 10^-10. That's what this function is supposed to do. Also, the type of this function does indeed constrain exponents to being integers: https://github.com/FormidableLabs/renature/blob/master/src/core/Math.rei#L7 In this definition, `exp` is defined as _having_ to be an `int`.

Reply:
Noice!